pkg/steps/pre_publish_images.go

```diff
@@ -0,0 +1,189 @@
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	imageapi "github.com/openshift/api/image/v1"
+	"github.com/openshift/ci-operator/pkg/api"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PrePublishOutputImageTagStep will ensure that a tag exists
+// in the named ImageStream that resolves to the built
+// pipeline image
+type prePublishOutputImageTagStep struct {
+	config    api.PrePublishOutputImageTagStepConfiguration
+	istClient imageclientset.ImageStreamTagsGetter
+	isClient  imageclientset.ImageStreamsGetter
+	jobSpec   *api.JobSpec
+}
+
+type prepublishConfig struct {
+	namespace, name, tag string
+}
+
+func (s *prePublishOutputImageTagStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *prePublishOutputImageTagStep) getprePublishConfig() (prepublishConfig, error) {
+	if len(s.jobSpec.Refs.Pulls) == 0 {
+		return prepublishConfig{}, fmt.Errorf("did not find a pull request associated with job")
+	}
+	namespace := s.config.To.Namespace
+	if namespace == "" {
+		return prepublishConfig{}, fmt.Errorf("invalid prePublishOutputImageTagStep, namespace must be set")
```

Comment:
Yes, I wasn't sure if we should do that since none of the existing steps except e2e seem to fail at step init. It would definitely be better to error before run, if we can.

Reply:
Oh, you mean in `Validate()`? I didn't realize that validation was done outside the steps package.

Reply:
Oh, but validate isnt run for raw_steps. So are you thinking `defaults.go`?

Reply:
We could validate raw steps, not sure why we don't

Reply:
@stevekuznetsov I could add some beginnings of that probably. I'd be looping and validating raw steps that are of the the pre-publish type, ignoring the others.